arrow-avro/src/reader/record.rs

```diff
@@ -37,34 +36,43 @@ use std::sync::Arc;
 const DEFAULT_CAPACITY: usize = 1024;
 
 /// Decodes avro encoded data into [`RecordBatch`]
+#[derive(Debug)]
 pub struct RecordDecoder {
     schema: SchemaRef,
     fields: Vec<Decoder>,
     use_utf8view: bool,
+    strict_mode: bool,
 }
 
 impl RecordDecoder {
     /// Create a new [`RecordDecoder`] from the provided [`AvroDataType`] with default options
     pub fn try_new(data_type: &AvroDataType) -> Result<Self, ArrowError> {
-        Self::try_new_with_options(data_type, ReadOptions::default())
+        Self::try_new_with_options(data_type, false, false)
```

Comment:
The JSON reader takes an approach I like -- its [builder](https://docs.rs/arrow-json/latest/arrow_json/reader/struct.ReaderBuilder.html) can produce a reader or a decoder, but the option-wrangling -- including [with_strict_mode](https://docs.rs/arrow-json/latest/arrow_json/reader/struct.ReaderBuilder.html#method.with_strict_mode) -- is largely the same for both. 

Something to consider as a follow-up, perhaps?

Reply:
We could also make the configuration more self documenting like

```suggestion
        Self::new(data_type)
          .with_utf8_view(true), 
          ...
```